# House Price - Kaggle competition

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from sklearn.model_selection import train_test_split

In [2]:
# importing data
train_data = pd.read_csv('./train.csv')
upload_data = pd.read_csv('./test.csv')

## 1 - Missing data imputation

### 1.1 - Variables with missing

In [35]:
# percentage of missing in variables
missing_var = train_data.isnull().sum()/train_data.shape[0]
missing_var = missing_var[missing_var > 0]
missing_var

LotFrontage     0.177397
Alley           0.937671
MasVnrType      0.005479
MasVnrArea      0.005479
BsmtQual        0.025342
BsmtCond        0.025342
BsmtExposure    0.026027
BsmtFinType1    0.025342
BsmtFinType2    0.026027
Electrical      0.000685
FireplaceQu     0.472603
GarageType      0.055479
GarageYrBlt     0.055479
GarageFinish    0.055479
GarageQual      0.055479
GarageCond      0.055479
PoolQC          0.995205
Fence           0.807534
MiscFeature     0.963014
dtype: float64

### 1.2 - Missing variable's description

    LotFrontage     0.177397
Linear feet of street connected to property  
For not corner houses, could be 1/4 of the total area  
For corner houses, could be 1/2 of the total area  

    Alley           0.937671
    Alley: Type of alley access to property

           Grvl	Gravel
           Pave	Paved
           NA 	No alley access
Probably ok having missing here

    MasVnrType      0.005479
    MasVnrArea      0.005479
"Masonry veneer walls consist of a single non-structural external layer of masonry work, typically brick, backed by an air space."  
Could have missing
    
    BsmtQual        0.025342
    BsmtCond        0.025342
Could have missing  

    BsmtExposure    0.026027
    BsmtFinType1    0.025342
    BsmtFinType2    0.026027
    Electrical      0.000685
    FireplaceQu     0.472603
    GarageType      0.055479
    GarageYrBlt     0.055479
    GarageFinish    0.055479
    GarageQual      0.055479
    GarageCond      0.055479
    PoolQC          0.995205
    Fence           0.807534
    MiscFeature     0.963014